### ISSUE TITLE
feat(html): add support for declarative JSX elements

### ISSUE THREAD

**Issue by iamandrewluca** [storybookjs/storybook#11608]
Right now when exporting an element from a story we have to do this in an imperative way

```tsx
export const withEmoji = () => {
  const button = document.createElement('button');
  button.innerText = '😀 😎 👍 💯';
  return button;
};
```

Would be great if we could do this in an declarative way using JSX (events also work)

```tsx
export const withEmoji = () => {
  return <button>😀 😎 👍 💯</button>;
};
```

Current alternative for declarative exporting is to use string html, but in my opinion I think JSX exporting is better.

```tsx
export const withEmoji = () => {
  return `<button>😀 😎 👍 💯</button>`;
};
```

I can help implemet this feature

I worked on an [experimntal PR](https://github.com/storybookjs/storybook/pull/11072) in the past, that used React to render JSX to string.  
But now I think using [`dom-chef`](https://github.com/vadimdemedes/dom-chef) will be better, because it creates HTML elements instead of string template

**Comment by iamandrewluca** [comment]
This will be closed when #11634 will be merged or closed

**Comment by iamandrewluca** [comment]
@stale Still in progress :slightly_smiling_face:
